website/markdown/docs/contribution/generation-pipeline.mdx

```diff
@@ -0,0 +1,84 @@
+---
+name: Generation Pipeline
+excerpt: 'This document describes how the generation pipeline works.'
+---
+
+## Generation Pipeline
+
+The process of generating projects in Tuist has evolved over time and through several iterations. 
+It is now more akin to a pipeline that entails several steps each with a particular purpose and responsibility. 
+The aim is to keep each of those steps simple while still allowing more complex features to be built on top of them.
+
+## Terms
+
+- **Manifest**: Those are the public user facing files used to describe projects. (e.g. `Project.swift`, `Workspace.swift`, `Config.swift`, etc...)
+    - The types are declared in the `ProjectDescription` target
+    - Those types can't depend on any other targets/modules as the `ProjectDescription` target is shipped as a standalone dynamic library with each Tuist release
+- **Model**: Those are the internal model types used to represent various entities of the Tuist Graph and supporting types.
+    - The types are declared in the `TuistCore` target
+    - Many types are similar to the ones defined in the `ProjectDescription` target
+        - Model types however contain more information than their `ProjectDescription` counterpart (e.g. expanded paths)
+        - Model types are often optimized for simpler internal usage
+- **Graph**: This is a collection of Tuist **Model**s along with all their loaded dependency information in a graph representation
+    - This is the primary entity used to at the heart of many Tuist features including project generation
+- **Model Mapper**: This is a type used to transform individual **Model**s ahead of loading the graph
+    - There 3 different sub-types `WorkspaceMapper`, `ProjectMapper` and `TargetMapper` which can be used to transform the corresponding model type
+    - In addition to transforming models, mappers can also declare a set of **Side Effects Descriptors** to perform at the end of the generation pipeline
+    - The mappers themselves do not perform the side effects
+- **Graph Mapper**: This is a type used to transform the entire **Graph**
```

Comment:
maybe this can be added to `Model Mapper` section? The subcomments for this are almost identical.

Reply:
They are indeed similar, the graph mappers are a separate set of components though hence why it's called out separately. I've abbreviated it to reduce the repetition 👍

Reply:
looks good, thanks!